File: test/services/jv/src/main/java/example/PetStoreImpl.java
package example;

import petstore.api.Pet;
import petstore.api.PetStatus;

import java.util.Comparator;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicLong;

public final class PetStoreImpl implements petstore.api.PetStore {
    private final Map<Long, Pet> pets = new ConcurrentHashMap<>();
    private final AtomicLong nextId = new AtomicLong(1);

    public void reset() {
        pets.clear();
        nextId.set(1);
    }

    @Override
    public petstore.api.petstore.addpet.Out addPet(petstore.api.petstore.addpet.In arg) {
        long id = nextId.getAndIncrement();
        Pet pet = new Pet(id, arg.name(), arg.status(), arg.tag());
        pets.put(id, pet);
        return new petstore.api.petstore.addpet.Out(pet);
    }

    @Override
    public petstore.api.petstore.getpet.Out getPet(petstore.api.petstore.getpet.In arg) {
        Pet pet = pets.get(arg.id());
        if (pet == null) {
            throw new RuntimeException("Pet not found: " + arg.id());
        }
        return new petstore.api.petstore.getpet.Out(pet);
    }

    @Override
    public petstore.api.petstore.listpets.Out listPets(petstore.api.petstore.listpets.In arg) {
        List<Pet> sorted = pets.values().stream()
            .sorted(Comparator.comparingLong(Pet::id))
            .toList();
        return new petstore.api.petstore.listpets.Out(sorted);
    }

    @Override
    public petstore.api.petstore.deletepet.Out deletePet(petstore.api.petstore.deletepet.In arg) {
        Pet removed = pets.remove(arg.id());
        return new petstore.api.petstore.deletepet.Out(removed != null);
    }
}

File: test/services/kt/src/main/kotlin/example/PetStoreImpl.kt
package example

import java.util.concurrent.atomic.AtomicLong

class PetStoreImpl : petstore.api.PetStore {
    private val pets = mutableMapOf<Long, petstore.api.Pet>()
    private val nextId = AtomicLong(1)

    fun reset() {
        pets.clear()
        nextId.set(1)
    }

    override fun addPet(arg: petstore.api.petstore.addpet.In): petstore.api.petstore.addpet.Out {
        val id = nextId.getAndIncrement()
        val pet = petstore.api.Pet(
            id = id,
            name = arg.name,
            status = arg.status,
            tag = arg.tag
        )
        pets[id] = pet
        return petstore.api.petstore.addpet.Out(pet = pet)
    }

    override fun getPet(arg: petstore.api.petstore.getpet.In): petstore.api.petstore.getpet.Out {
        val pet = pets[arg.id] ?: throw NoSuchElementException("Pet not found: id=${arg.id}")
        return petstore.api.petstore.getpet.Out(pet = pet)
    }

    override fun listPets(arg: petstore.api.petstore.listpets.In): petstore.api.petstore.listpets.Out {
        val sorted = pets.values.sortedBy { it.id }
        return petstore.api.petstore.listpets.Out(pets = sorted)
    }

    override fun deletePet(arg: petstore.api.petstore.deletepet.In): petstore.api.petstore.deletepet.Out {
        val existed = pets.remove(arg.id) != null
        return petstore.api.petstore.deletepet.Out(deleted = existed)
    }
}
